### PR TITLE
Add dart/ts interoperability tests

### PR DESCRIPTION
**Pull request by kevindingens**
**Interoperability tests**
A suite of 5 tests (pass1-1, pass1-2, pass2-1, pass2-2, and pass3-1) with exact copies of each in the dart SDK, meant to run serially and alternating between the two SDKS (ts executes pass1-1, then dart executes pass1-2, then ts pass2-1, dart pass2-2, etc). Can be tested with/without encryption and using dialect cloud/solana backends
**Utility tests**
A token test used to output an auth token to paste into the respective dart test, and to test a token pasted from the respective dart test.
**Sorting**
I added message sorting to this SDK so I could test ordering in the interoperability tests. Happy to redact
**Data service api updates**
Every data service call returned 404 until I removed the "api" path from each data service api endpoint. Also happy to redact

### CODE REVIEW

File: src/internal/data-service-api/data-service-dialects-api.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class DataServiceDialectsApiClient implements DataServiceDialectsApi {
     const token = await this.tokenProvider.get();
     return withReThrowingDataServiceError(
       axios
-        .post<DialectAccountDto>(`${this.baseUrl}/api/v1/dialects`, command, {
+        .post<DialectAccountDto>(`${this.baseUrl}/v0/dialects`, command, {
```
Comment (kevindingens) [inline src/internal/data-service-api/data-service-dialects-api.ts:33]: Changing these URLs was the only way I could get the tests to work - let me know if you want to change these back @tsmbl

Comment (tsmbl) [reply src/internal/data-service-api/data-service-dialects-api.ts:33]: @kevindingens, seems like you have old data service version, should be v1 here
Could you please verify that you run last data-service version from main branch?

Comment (kevindingens) [reply src/internal/data-service-api/data-service-dialects-api.ts:33]: Yes I pulled late last night to test and I pulled a few minutes ago. Same thing. Here is a screenshot of the Swagger
Screen Shot 2022-06-23 at 10 17 02 AM

**Comment by tsmbl** [comment]
Closing since dart is not relevant anymore, will keep the branch

**Pull request closed without merging**